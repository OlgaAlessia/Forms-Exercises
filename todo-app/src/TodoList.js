import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuid } from "uuid";

function TodoList() {
    const [todos, setTodos] = useState([]);
    //const [isCompleted, setIsCompleted] = useState(false);


    function handleRemove(id) {
        setTodos(todos.filter(todo => todo.id !== id));
    }

    /*function handleCompleted() {
        setIsCompleted(isCompleted = !isCompleted);
    }*/

    /** Add new box object to boxList. */
    const addTodo = todo => {
        let newtodo = { ...todo, id: uuid() };
        setTodos(todos => [...todos, newtodo]);
    };
    // end addBox

    return (
        <div>
            <h3>Todo List</h3>
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList">
                {todos.map(({ id, task }) =>

                    <Todo key={id} task={task}
                        handleRemove={evt => handleRemove(id)}
                    />
                )}
            </div>
        </div>
    );
    /* handleCompleted={evt => handleCompleted()}
                        isCompleted={isCompleted} */
};
// end

export default TodoList;
