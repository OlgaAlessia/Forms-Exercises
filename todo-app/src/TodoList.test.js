import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new Todo", function () {
  const { queryByText, getByLabelText } = render(<TodoList />);
  const input = getByLabelText("Task:");
  const btn = queryByText("Add a Todo!");
  expect(queryByText('Walk the dog')).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: 'Walk the dog' } });
  fireEvent.click(btn);
  expect(queryByText('Walk the dog')).toBeInTheDocument();
})


