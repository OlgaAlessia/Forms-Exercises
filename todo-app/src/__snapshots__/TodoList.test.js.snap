// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form
      class="NewBoxForm"
    >
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <br />
      <button>
        Add a Todo!
      </button>
    </form>
    <div
      class="TodoList"
    />
  </div>
</DocumentFragment>
`;
